integritybackend/actions.py

```diff
@@ -44,121 +42,185 @@ def archive2(self, zip_path: str, org_config: dict, collection_id: str):
-            )
-        if f"{content_sha_unverified}-meta-recorder.json" not in zip_listing:
-            raise Exception(
-                f"ZIP at {zip_path} has no recorder metadata file: {zip_listing}"
-            )
-
-        # Extract content file
-        tmp_dir = asset_helper.path_for_action_tmp(collection_id, action_name)
-        zip_dir = os.path.join(tmp_dir, content_sha_unverified)
-        extracted_content = os.path.join(zip_dir, content_filename)
-        file_util.create_dir(zip_dir)
-        zip_util.extract_file(tmp_zip, content_filename, extracted_content)
-
-        # Generate content hashes and verify
-        content_sha = file_util.digest_sha256(extracted_content)
-        if content_sha != content_sha_unverified:
-            raise Exception(f"SHA-256 of content does not match file name: {zip_path}")
-        content_cid = file_util.digest_cidv1(extracted_content)
-        content_md5 = file_util.digest_md5(extracted_content)
-        _logger.info("Content verified for archival: {zip_path}")
-
-        # Register on OpenTimestamp and add that file to zip
-        if action_params["registration_policies"]["opentimestamps"]["active"]:
-            content_ots = extracted_content + ".ots"
-            file_util.register_timestamp(extracted_content, content_ots)
-            zip_util.append(
-                tmp_zip,
-                content_ots,
-                "proofs/" + os.path.basename(content_ots),
-            )
-            _logger.info(f"Content registered on opentimestamps: {content_ots}")
-        else:
-            _logger.info(f"Content registration on opentimestamps skipped")
-
-        # Get archive ZIP hashes
-        zip_sha = file_util.digest_sha256(tmp_zip)
-        zip_md5 = file_util.digest_md5(tmp_zip)
-        zip_cid = file_util.digest_cidv1(tmp_zip)
-
-        # Rename archive zip to SHA-256 of itself
-        archive_zip = os.path.join(archive_dir, zip_sha + ".zip")
-        os.rename(tmp_zip, archive_zip)
-        _logger.info(f"Archive zip generated: {archive_zip}")
-
-        # Encrypt archive ZIP
-        aes_key = crypto_util.get_key(action_params["encryption"]["key"])
-        tmp_encrypted_zip = os.path.join(archive_dir, zip_sha + ".encrypted")
-        file_util.encrypt(aes_key, archive_zip, tmp_encrypted_zip)
-
-        # Get encrypted ZIP hashes
-        enc_zip_sha = file_util.digest_sha256(tmp_encrypted_zip)
-        enc_zip_md5 = file_util.digest_md5(tmp_encrypted_zip)
-        enc_zip_cid = file_util.digest_cidv1(tmp_encrypted_zip)
-
-        # Rename encrypted ZIP to SHA-256 of itself
-        encrypted_zip = os.path.join(archive_dir, enc_zip_sha + ".encrypted")
-        os.rename(tmp_encrypted_zip, encrypted_zip)
-        _logger.info(f"Encrypted zip generated: {encrypted_zip}")
-
-        # TODO: Register encrypted ZIP on Numbers Protocol
-        # TODO: Register encrypted ZIP on ISCN
-        # Iscn.register_archive(path)
+        try:
```

Comment:
Could we have `_archive` and `archive` or something to avoid putting all the code in a try block? Would make things more readable.